src/mach/compact_unwind_info.rs

```diff
@@ -0,0 +1,1503 @@
+                println!("    Second level index[{}]: offset in section=0x{:08x}, base function=0x{:08x}",
+                iter.first_idx,
+                first.second_level_page_offset,
+                first.first_address);
+            }
+
+            // Dump the entry
+
+            // Feed in own instruction_address as a dummy value (we don't need it for this format)
+            let entry =
+                iter.complete_entry(&raw_entry, raw_entry.instruction_address, &first, &second)?;
+            if let OpcodeOrIndex::Index(opcode_idx) = raw_entry.opcode_or_index {
+                println!(
+                    "      [{}]: function offset=0x{:08x}, encoding[{}]=0x{:08x}",
+                    second_idx, entry.instruction_address, opcode_idx, entry.opcode.0
+                );
+            } else {
+                println!(
+                    "      [{}]: function offset=0x{:08x}, encoding=0x{:08x}",
+                    second_idx, entry.instruction_address, entry.opcode.0
+                );
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl SecondLevelPage {
+    fn len(&self) -> u32 {
+        match *self {
+            SecondLevelPage::Regular(ref page) => page.entries_len as u32,
+            SecondLevelPage::Compressed(ref page) => page.entries_len as u32,
+        }
+    }
+}
+
+impl CompactUnwindInfoEntry {
+    /// Gets cfi instructions associated with this entry.
+    pub fn cfi_instructions(
+        &self,
+        iter: &CompactUnwindInfoIter,
+    ) -> Option<impl Iterator<Item = CfiOp>> {
+        self.opcode.cfi_instructions(iter)
+    }
+}
+
+// Arch-generic stuff
+impl Opcode {
+    fn cfi_instructions(
+        &self,
+        iter: &CompactUnwindInfoIter,
+    ) -> Option<impl Iterator<Item = CfiOp>> {
+        match iter.arch {
+            Arch::X86 | Arch::X64  => {
+                self.x86_cfi_instructions(iter)
+            }
+            Arch::Arm64 => {
+                self.arm64_cfi_instructions(iter)
+            }
+            _ => None,
+        }
+    }
+
+    fn pointer_size(&self, iter: &CompactUnwindInfoIter) -> u32 {
+        match iter.arch {
+            Arch::X86 => 4,
+            Arch::X64 => 8,
+            Arch::Arm64 => 8,
+            _ => unimplemented!(),
+        }
+    }
+
+    /*
+    // potentially needed for future work:
+
+    fn is_start(&self) -> bool {
+        let offset = 32 - 1;
+        (self.0 & (1 << offset)) != 0
+    }
+    fn has_lsda(&self) -> bool{
+        let offset = 32 - 2;
+        (self.0 & (1 << offset)) != 0
+    }
+    fn personality_index(&self) -> u32 {
+        let offset = 32 - 4;
+        (self.0 >> offset) & 0b11
+    }
+    */
+}
+
+// x86/x64 implementation
+impl Opcode {
+    fn x86_cfi_instructions(
+        &self,
+        iter: &CompactUnwindInfoIter,
+    ) -> Option<std::vec::IntoIter<CfiOp>> {
+        let pointer_size = self.pointer_size(iter) as i32;
+        // TODO: don't allocate for this (use ArrayVec..?)
```

Comment:
I'd prefer to take on no more dependencies

Reply:
Ok! In the original PR on symbolic I have now replaced the Vec with a little adhoc ArrayVec impl to get rid of the Vec without introducing that dependency.